test/LendingPool.t.sol

```diff
@@ -368,6 +368,47 @@ contract DepositAndWithdrawalTest is LendingPoolTest {
         pool.depositInLendingPool(amount1, liquidityProvider);
     }
 
+    function testRevert_depositInLendingPool_SupplyCap(uint256 amount) public {
+        // Given: amount should be greater than 1
+        vm.assume(amount > 1);
+
+        // When: supply cap is set to 1
+        vm.prank(creator);
+        pool.setSupplyCap(1);
```

Comment:
fuzz the supplycap, and set a vm.assume(pool.realizedLiq + amount > supplycap), without it it wouldn't have caught the totalsupply vs realisedliquidity issue